### PR TITLE
GEODE-5533 Support Gradle 4.9 publication enhancements

### PR DESCRIPTION
**Pull request by rhoughton-pivot**
Upgrading to Gradle 4.9 broke publish/install due to a non-existent jar
output from geode-assembly. More diligent pruning of the archives
configuration has fixed the problem.

Co-authored-by: Robert Houghton <rhoughton@pivotal.io>
Co-authored-by: Patrick Rhomberg <prhomberg@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: gradle/java.gradle [hunk 1]
```diff
@@ -23,7 +23,6 @@ allprojects {
 }
 
 subprojects {
-  apply plugin: 'java'
```
Comment (metatype) [inline gradle/java.gradle]: Why is this removed?

File: geode-assembly/build.gradle [hunk 2]
```diff
@@ -29,10 +27,17 @@ extraArchive {
...
 
-// Gradle doesn't automatically remove the jar artifact even though we disabled it
-// this causes publishing to fail.  So we nuke all the disabled artifacts from all configurations.
+logger.info("Gradle doesn't automatically remove the jar artifact even though we disabled it")
+logger.info("this causes publishing to fail.  So we nuke all the disabled artifacts from all configurations.")
 configurations.all {
-  artifacts.removeAll artifacts.findAll { it instanceof ArchivePublishArtifact && !it.archiveTask.enabled }
+  def configName = it.name
+  artifacts.each {
+    logger.info("Pruning configName = " + configName + " name=" + it.name + " extension=" + it.extension + " classifier=" + it.classifier)
```
Comment (metatype) [inline geode-assembly/build.gradle]: Should this be `debug` level instead of `info`?

File: geode-assembly/build.gradle [hunk 3]
```diff
@@ -29,10 +27,17 @@ extraArchive {
...
+  }
+  artifacts.removeAll{ it.name == jar.baseName && it.extension == 'jar' && it.classifier == jar.classifier }
```
Comment (metatype) [inline geode-assembly/build.gradle]: Is one of these conditions sufficient?

File: gradle/java.gradle (hunk 1 above)
Comment (rhoughton-pivot) [reply gradle/java.gradle]: Duplicate of line 19, where it is applied via `allprojects{}`, which includes the root project.

File: geode-assembly/build.gradle (hunk 3 above)
Comment (rhoughton-pivot) [reply geode-assembly/build.gradle]: No, or else you could remove a mis-named artifact, or the `tar` or `zip` artifacts, or accidentally add a `test` or `integrationTest` artifact

File: geode-assembly/build.gradle (hunk 2 above)
Comment (rhoughton-pivot) [reply geode-assembly/build.gradle]: As debug info it was lost in the noise, and logging instead of comments seems more-sane to me.

**Comment by rhoughton-pivot** [comment]
@upthewaterspout info is a non-default flag that would need to be set to get the output

**Comment by pivotal-jbarrett** [comment]
@rhoughton-pivot Since this was only caught by doing `install` let's add `install` to the CI pipeline. Perhaps as part of the `Build` job.

**Comment by rhoughton-pivot** [comment]
@pivotal-jbarrett Added
